src/CloudNative.CloudEvents/BinaryDataUtilities.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright 2021 Cloud Native Foundation.
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace CloudNative.CloudEvents
+{
+    /// <summary>
+    /// Utilities methods for dealing with binary data, converting between
+    /// streams, arrays, Memory{T} etc.
+    /// </summary>
+    internal static class BinaryDataUtilities
+    {
+        internal async static Task<byte[]> ToByteArrayAsync(Stream stream)
+        {
+            // TODO: Optimize if it's already a MemoryStream?
+            var memory = new MemoryStream();
```

Comment:
should there be any validation around the size of the stream if it is seekable?

Reply:
I don't think so. If someone chooses to pass in an enormous stream that will eat up all their memory, that's their call - I suspect any validation we'd put in place would have to be entirely arbitrary.